util/system/path.go

```diff
@@ -16,3 +25,180 @@ func DefaultPathEnv(os string) string {
 	}
 	return DefaultPathEnvUnix
 }
+
+// NormalizePath functions similarly to filepath.Join(), but with a few distinctions:
+//   - use the platform flag to decide how we clean the path
+//   - always return an absolute path
+//   - always strip drive letters for Windows paths
+//   - optionally keep the trailing slashes on paths
+//   - paths are returned using forward slashes
+func NormalizePath(parent, path, platform string, keepSlash bool) (string, error) {
+	origPath := path
+
+	if parent == "" {
+		parent = "/"
+	}
+
+	var err error
+	parent, err = CheckSystemDriveAndRemoveDriveLetter(parent, platform)
+	if err != nil {
+		return "", errors.Wrap(err, "removing drive letter")
+	}
+
+	if !IsAbs(parent, platform) {
+		parent = filepath.Join("/", parent)
+	}
+
+	if path == "" {
+		// New workdir is empty. Use the "current" workdir. It should already
+		// be an absolute path.
+		path = parent
+	}
+
+	path, err = CheckSystemDriveAndRemoveDriveLetter(path, platform)
+	if err != nil {
+		return "", errors.Wrap(err, "removing drive letter")
+	}
+
+	if !IsAbs(path, platform) {
```

Comment:
This doesn't look like a condition in `path.Join`

Reply:
It's not. This function is not meant to be a drop-in replacement for ```filepath.Join()```. It's meant to be used as a replacement for [```normalizePath()```](https://github.com/moby/buildkit/blob/master/client/llb/fileop.go#L772-L789), but is also useful in other cases.

Reply:
This function does the right thing in situations where on Widows, using ```filepath.Join("/", somePath)``` (used in several places) would yield: ```\C:\somePath```, making the path invalid.